packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -62,6 +63,149 @@ const TextStyle _kToolbarButtonDisabledFontStyle = TextStyle(
 // Eyeballed value.
 const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(vertical: 10.0, horizontal: 18.0);
 
+// Generates the child that's passed into CupertinoTextSelectionToolbar.
+class _CupertinoTextSelectionToolbarWrapper extends StatefulWidget {
+  const _CupertinoTextSelectionToolbarWrapper({
+    Key key,
+    this.arrowTipX,
+    this.barTopY,
+    this.clipboardStatus,
+    this.handleCut,
+    this.handleCopy,
+    this.handlePaste,
+    this.handleSelectAll,
+    this.isArrowPointingDown,
+  }) : super(key: key);
+
+  final double arrowTipX;
+  final double barTopY;
+  final ClipboardStatusNotifier clipboardStatus;
+  final VoidCallback handleCut;
+  final VoidCallback handleCopy;
+  final VoidCallback handlePaste;
+  final VoidCallback handleSelectAll;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarWrapperState createState() => _CupertinoTextSelectionToolbarWrapperState();
+}
+
+class _CupertinoTextSelectionToolbarWrapperState extends State<_CupertinoTextSelectionToolbarWrapper> {
+  ClipboardStatusNotifier _clipboardStatus;
+
+  void _onChangedClipboardStatus() {
+    setState(() {
+      // Inform the widget that the value of clipboardStatus has changed.
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _clipboardStatus = widget.clipboardStatus ?? ClipboardStatusNotifier();
```

Comment:
Wondering if we could simplify this a little bit by requiring that you pass a non-null ClipboardStatusNotifier to the TextSelectionOverlay? That should simplify the logic in didUpdateWidget a little bit.

Reply:
I avoided that because I thought it would be a breaking change.  What do you think?

Reply:
Do we have customers that implement their own TextSelectionOverlay?

Reply:
I'm not sure.  The text selection menu has been pretty inflexible, but I think people might be overriding TextSelectionControls and not TextSelectionOverlay when they want things like custom buttons.  I see nothing internally that directly uses TextSelectionOverlay.

@HansMuller Do you have any opinion on breaking TextSelectionOverlay?

Reply:
If we're going to break TextSelectionOverlay, I think doing so should be part of a carefully considered plan about how to make the text selection menu (easily) extensible.